be/src/olap/field.h

```diff
@@ -389,6 +397,7 @@ class VarcharField: public Field {
         return  _length - OLAP_STRING_MAX_BYTES;
     }
 
+    //todo(wangbo) using _length replace (_length - OLAP_STRING_MAX_BYTES) when segment v2 release
```

Comment:
I think we can't replace _length, because all _length has been written in tablet metadata. You can comment why we minus OLAP_STRING_MAX_BYTES. Because when v1 is deleted, other people will feel strange why we minus OLAP_STRING_MAX_BYTES.

Reply:
for v2 varchar,If allocate string value'length < field.length,then write a varchar which length is field.length into the allocated value ,this won't cause wrong value?

Reply:
In your case it would cause problem. But for v2 max length of a varchar field is also field.length - OLAP_STRING_MAX_BYTES. It's impossible to write a longer value to allocated field.

Reply:
👌;allocate_memory is used by max/min value and row_cursor used to write value
I checked latter case, varchar which allocated by Field.allocate_memory is written by direct copy,so the length here doesn't matter